### PR TITLE
Fix #938

### LINKED ISSUE #938: TraversableLike: error: local definition of $anon escapes
```
$ dotc ./scala-scala/src/library/scala/collection/TraversableLike.scala
warning: encountered F-bounded higher-kinded type parameters for type CC; assuming they are invariant
warning: encountered F-bounded higher-kinded type parameters for type CC; assuming they are invariant
warning: encountered F-bounded higher-kinded type parameters for type CC; assuming they are invariant
warning: encountered F-bounded higher-kinded type parameters for type CC; assuming they are invariant
./scala-scala/src/library/scala/collection/TraversableLike.scala:666: error: local definition of $anon escapes as part of expression's type scala.collection.TraversableView{A =+ A; Coll =+ Repr; underlying: Repr}
  def view = new TraversableView[A, Repr] {
                 ^
four warnings found
one error found
```

### PR DESCRIPTION
**Pull request by odersky**
Review by @smarter. And, maybe take it over if there are still things to do.

### CODE REVIEW

File: src/dotty/tools/dotc/core/TypeComparer.scala [hunk 1]
```diff
@@ -488,6 +488,9 @@ class TypeComparer(initctx: Context) extends DotClass with ConstraintHandling {
         case _ => tp2 isRef ObjectClass
       }
       compareJavaArray
+    case tp1: ExprType if ctx.phase.id >= ctx.gettersPhase.id =>
```
Comment (smarter) [inline src/dotty/tools/dotc/core/TypeComparer.scala]: Should this be `ctx.phase.id > ctx.gettersPhase.id` like in https://github.com/dotty-staging/dotty/blob/fix-%23938/src/dotty/tools/dotc/core/TypeComparer.scala#L455 ?

Comment (odersky) [inline src/dotty/tools/dotc/core/TypeComparer.scala]: Yes, good catch.

**Comment by smarter** [comment]
LGTM.

**Comment by odersky** [comment]
Rebased to master

**Pull request merged**